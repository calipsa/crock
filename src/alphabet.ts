export default '0123456789abcdefghjkmnpqrstvwxyz'
