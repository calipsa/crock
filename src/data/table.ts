import alphabet from './alphabet'

// tslint:disable-next-line:interface-over-type-literal
type Table = {
  [letter: string]: number,
}

const alias: Table = {
  o: 0,
  i: 1,
  l: 1,
}

const table: Table = {}
// Invert 'alphabet'
for (let i = 0; i < alphabet.length; ++i) {
  const letter = alphabet[i]
  table[letter] = i
}

export default Object.freeze({
  ...table,
  ...alias,
})
