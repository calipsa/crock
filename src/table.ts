import alphabet from './alphabet'

// tslint:disable-next-line:interface-over-type-literal
type Table = {
  [letter: string]: number,
}

const alias = {
  o: 0,
  i: 1,
  l: 1,
}

const table: Table = {}
// Invert 'alphabet'
let i = 0
for (const letter of alphabet) {
  table[letter] = i++
}
// Splice in 'alias'
for (const [key, val] of Object.entries(alias)) {
  table[key] = table[val]
}

export default Object.freeze(table)
