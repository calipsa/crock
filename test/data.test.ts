import alphabet from '../src/data/alphabet'
import table from '../src/data/table'

describe('Testing data', () => {
  it('Should have the same alphabet', () => {
    expect(alphabet).toMatchSnapshot()
  })

  it('Should have the same table', () => {
    expect(table).toMatchSnapshot()
  })
})
