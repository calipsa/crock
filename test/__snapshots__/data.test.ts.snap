// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing data Should have the same alphabet 1`] = `"0123456789abcdefghjkmnpqrstvwxyz"`;

exports[`Testing data Should have the same table 1`] = `
Object {
  "0": 0,
  "1": 1,
  "2": 2,
  "3": 3,
  "4": 4,
  "5": 5,
  "6": 6,
  "7": 7,
  "8": 8,
  "9": 9,
  "a": 10,
  "b": 11,
  "c": 12,
  "d": 13,
  "e": 14,
  "f": 15,
  "g": 16,
  "h": 17,
  "i": 1,
  "j": 18,
  "k": 19,
  "l": 1,
  "m": 20,
  "n": 21,
  "o": 0,
  "p": 22,
  "q": 23,
  "r": 24,
  "s": 25,
  "t": 26,
  "v": 27,
  "w": 28,
  "x": 29,
  "y": 30,
  "z": 31,
}
`;
